bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/BaseHomeThingHandler.java

```diff
@@ -30,7 +30,9 @@ public BaseHomeThingHandler(Thing thing) {
     }
 
     public Long getHomeId() {
-        return getHomeHandler().getHomeId();
+        // null pointer exception fixed by Andrew Fiddian-Green
+        TadoHomeHandler handler = getHomeHandler();
+        return handler != null ? handler.getHomeId() : new Long(0);
```

Comment:
```suggestion
        return handler != null ? handler.getHomeId() : 0L;
```

Reply:
Ok, I will make this change. But for the avoidance of doubt, the method return type is "Long" rather than "long", so I want to be sure to return an actual Long, in order to avoid messing too deeply in the Original Author's code..

Reply:
I reverted to my original code below; the reason is that your proposal ( `0L` ) causes the binding to crash :(

`return handler != null ? handler.getHomeId() : new Long(0);`